tests/unit/lms/services/canvas_api/basic_test.py

```diff
@@ -0,0 +1,177 @@
+from unittest.mock import call, create_autospec
+
+import pytest
+from h_matchers import Any
+
+from lms.services import CanvasAPIError
+from lms.services.canvas_api import BasicClient
+from lms.validation import RequestsResponseSchema, ValidationError
+from tests import factories
+
+
+@pytest.mark.usefixtures("http_session")
+class TestBasicClient:
```

Comment:
There's only one public method nowadays, so the tests here all try to corner one behavior of it